module/applications/actor/base-sheet.mjs

```diff
@@ -251,6 +251,26 @@ export default class ActorSheet5e extends ActorSheet {
     return tags;
   }
 
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare preparation warnings for display.
+   * @param {Array<string|object>} warnings  Array of warnings.
+   * @returns {object[]}                     Prepared warnings.
+   * @protected
+   */
+  _getWarnings(warnings) {
+    const prepared = [];
+    for ( const warning of warnings ) {
+      if ( typeof warning === "string" ) {
```

Comment:
Do you foresee any issues with converting all of our preparation warnings into these richer objects so we don't need two separate branches when handling them?